compose.override.dist.yaml

```diff
@@ -4,14 +4,14 @@ services:
   php-7.4: &php
     build:
       target: php
-      args: &base_build_args
+      args:
         DOCKER_USER_ID: ${DOCKER_USER_ID:-1000}
         DOCKER_GROUP_ID: ${DOCKER_GROUP_ID:-1000}
     user: ${DOCKER_USER_ID:-1000}:${DOCKER_GROUP_ID:-1000}
     extra_hosts:
-      # Required for Docker Linux until natively supported.
-      # See https://github.com/docker/for-linux/issues/264
-      - 'host.docker.internal: 172.17.0.1'
+      # Allow container to connect to host to use XDebug step debugging.
+      # This is not needed when using Docker Desktop as it supports this natively.
+      - 'host.docker.internal:host-gateway'
```

Comment:
What is the actual compatibility of this? I get `Error response from daemon: invalid IP address in add-host: " host-gateway"` when I try to use it on Mac with OrbStack (with `docker compose run php-8.3 xdebug php-cs-fixer check --verbose --diff`).

Reply:
Initially I pushed a commit with a space in the value:
```
- 'host.docker.internal: host-gateway'
```
which is invalid. I quickly removed it and force-pushed. Is is possible that you tried this change right before I removed the space?

I don't know OrbStack but it's possible it doesn't support `host-gateway`. However, reading the docs it seems adding the extra host is not needed at all as [`host.internal.docker` is natively supported](https://docs.orbstack.dev/docker/network#connecting-to-servers-on-mac) like with Docker Desktop. Do you confirm it works? If so, maybe I should rephrase the comment to mention OrbStack as well.

Reply:
Yes, it's possible, it could be also auto-formatted when pasting it to actual `comppse.override.yaml` which I already had. I'll re-check it later 🙂.

Reply:
It works without superfluous space 👍.